circuit/primitives/src/event_signature.rs

```diff
@@ -170,4 +237,89 @@ pub mod tests {
+        )
+    }
+    #[test]
+    fn successfully_decodes_dfd_for_3_parallel_events_in_single_brackets() {
+        let valid_dfd = "(A,B,C)";
+        // Important! If using .encode() instead of .as_bytes() + .to_vec(),
+        //  SCALE adds additional byte "92" to event name
+        let decode_res = decode_dfd(valid_dfd.as_bytes().to_vec());
+        assert_eq!(
+            decode_res,
+            Ok(vec![vec![b"A".to_vec(), b"B".to_vec(), b"C".to_vec(),]])
+        )
+    }
+    #[test]
+    fn successfully_decodes_dfd_for_3_parallel_events_in_triple_brackets() {
+        let valid_dfd = "(((A,B,C)))";
+        // Important! If using .encode() instead of .as_bytes() + .to_vec(),
+        //  SCALE adds additional byte "92" to event name
+        let decode_res = decode_dfd(valid_dfd.as_bytes().to_vec());
+        assert_eq!(
+            decode_res,
+            Ok(vec![vec![b"A".to_vec(), b"B".to_vec(), b"C".to_vec(),]])
+        )
+    }
+    #[test]
+    fn fails_to_decode_dfd_for_3_parallel_events_with_incorrect_closing_brackets() {
+        let valid_dfd = "(A,B,C";
+        // Important! If using .encode() instead of .as_bytes() + .to_vec(),
+        //  SCALE adds additional byte "92" to event name
+        let decode_res = decode_dfd(valid_dfd.as_bytes().to_vec());
+        assert_eq!(decode_res, Err("DFD Decoder - too many opening brackets"))
+    }
+    #[test]
+    fn fails_to_decode_dfd_for_3_parallel_events_with_incorrect_opening_brackets() {
+        let valid_dfd = "A,B,C)";
+        // Important! If using .encode() instead of .as_bytes() + .to_vec(),
+        //  SCALE adds additional byte "92" to event name
+        let decode_res = decode_dfd(valid_dfd.as_bytes().to_vec());
+        assert_eq!(
+            decode_res,
+            Err("DFD Decoder - attempt to edit step at incorrect depth")
+        )
+    }
+
+    #[test]
+    fn successfully_decodes_dfd_for_3_sequential_events() {
+        let valid_dfd = "(A(B(C)))";
+        // Important! If using .encode() instead of .as_bytes() + .to_vec(),
+        //  SCALE adds additional byte "92" to event name
+        let decode_res = decode_dfd(valid_dfd.as_bytes().to_vec());
+        assert_eq!(
+            decode_res,
+            Ok(vec![
+                vec![b"A".to_vec()],
+                vec![b"B".to_vec()],
+                vec![b"C".to_vec()]
+            ])
+        )
+    }
+
+    #[test]
+    fn successfully_decodes_dfd_for_2_sequential_events_after_2_parallel() {
+        let valid_dfd = "(D(C(A,B)))";
+        // Important! If using .encode() instead of .as_bytes() + .to_vec(),
+        //  SCALE adds additional byte "92" to event name
+        let decode_res = decode_dfd(valid_dfd.as_bytes().to_vec());
+        assert_eq!(
+            decode_res,
+            Ok(vec![
+                vec![b"D".to_vec()],
+                vec![b"C".to_vec()],
+                vec![b"A".to_vec(), b"B".to_vec()]
```

Comment:
Same here, should be reverse

Reply:
For parallel events won't make a difference